app/modules/adc.c

```diff
@@ -24,10 +24,49 @@ static int adc_readvdd33( lua_State* L )
   return 1;
 }
 
+// Lua: adc.force_init_byte107(x)
+static int adc_init107( lua_State *L )
+{
+  uint8_t byte107 = luaL_checkinteger (L, 1);
+
+  uint32 init_sector = flash_safe_get_sec_num () - 4;
+
+  // Note 32bit alignment so we can safely cast to uint32 for the flash api
+  char init_data[SPI_FLASH_SEC_SIZE] __attribute__((aligned(4)));
+
+  if (SPI_FLASH_RESULT_OK != flash_safe_read (
+    init_sector * SPI_FLASH_SEC_SIZE,
+    (uint32 *)init_data, sizeof(init_data)))
+      return luaL_error(L, "flash read error");
+
+  // If it's already the correct value, we don't need to force it
+  if (init_data[107] == byte107)
+  {
+    lua_pushboolean (L, false);
+    return 1;
+  }
+
+  // Nope, it differs, we need to rewrite it
+  init_data[107] = byte107;
+  if (SPI_FLASH_RESULT_OK != flash_safe_erase_sector (init_sector))
+    return luaL_error(L, "flash erase error");
+
+  if (SPI_FLASH_RESULT_OK != flash_safe_write (
+    init_sector * SPI_FLASH_SEC_SIZE,
+    (uint32 *)init_data, sizeof(init_data)))
+      return luaL_error(L, "flash write error");
+
+  lua_pushboolean (L, true);
+  return 1;
+}
+
 // Module function map
 static const LUA_REG_TYPE adc_map[] = {
   { LSTRKEY( "read" ),      LFUNCVAL( adc_sample ) },
-  { LSTRKEY( "readvdd33" ), LFUNCVAL( adc_readvdd33) },
+  { LSTRKEY( "readvdd33" ), LFUNCVAL( adc_readvdd33 ) },
+  { LSTRKEY( "force_init_byte107" ), LFUNCVAL( adc_init107 ) },
```

Comment:
I see your point, haven't thought of that, but I still think that detail shouldn't leak into Lua land. It's IMHO something for the docs.

> If anyone is using custom init_data blocks for RF tuning, I'd prefer them to be easily aware that they're mucking with it when/if they're using this ADC function.

I have no idea what "RF tuning" is or what RF stands for (radio frequency?) but how many use cases or users are we talking about here? One in a thousand?

As I know your reasoning now I'm undecided as well. Is it better to leak an implementation detail into the API or into the docs? Dunno. How likely is it that it's not going to be byte 107 anymore in the future?